### PR TITLE
Henry/new line clear

### PR DESCRIPTION
**Pull request by henrytran28**
Removes new lines when it is full and drops the rest of the lines to fill the empty lines.

### CODE REVIEW

**Comment by henrytran28** [comment]
@ctlai95 Please review when you have time

File: src/board/board.py [hunk 1]
```diff
@@ -38,35 +36,74 @@ def render_board(self):
...
+            square.render_square()
 
         # Render the ghost tetromino
         self.ghost_tetromino.render_tetromino()
 
         # Render current playable tetromino
         self.current_tetromino.render_tetromino()
 
+    def get_filled_indices(self):
+        """Returns the number of lines filled"""
```
Comment (ctlai95) [inline src/board/board.py]: Can you update these new functions to match `pydocstyle` please? See #28 for examples

File: src/board/board.py [hunk 2]
```diff
@@ -38,35 +36,74 @@ def render_board(self):
...
+        filled_indices = []
+        for j in range(self.height):
+            is_filled = True
```
Comment (ctlai95) [inline src/board/board.py]: let's rename this to just `filled`

File: src/board/board.py [hunk 3]
```diff
@@ -38,35 +36,74 @@ def render_board(self):
...
+                    board_tetrominos_squares_copy.remove(square)
+
+        self.board_tetrominos_squares = board_tetrominos_squares_copy
+
+    def drop_lines(self, indices):
+        """Drops the lines based on the given indices"""
+        lines_dropped = 0
+        cond = False
+        for index in indices:
+            log.debug("Current loop index: {}".format(index))
```
Comment (ctlai95) [inline src/board/board.py]: We can probably remove this, or print a better debug message

File: src/board/board.py [hunk 4]
```diff
@@ -38,35 +36,74 @@ def render_board(self):
...
+            for square in self.board_tetrominos_squares:
+                if square.y == index:
+                    board_tetrominos_squares_copy.remove(square)
+
+        self.board_tetrominos_squares = board_tetrominos_squares_copy
+
+    def drop_lines(self, indices):
+        """Drops the lines based on the given indices"""
+        lines_dropped = 0
+        cond = False
```
Comment (ctlai95) [inline src/board/board.py]: I don't think this boolean is even needed lol. 
```
lines_dropped = 0
for index in indices:
    for square in self.board_tetrominos_squares:
        if square.y > index - lines_dropped:
            square.y = square.y - 1
    lines_dropped += 1
```

File: src/board/board_test.py [hunk 5]
```diff
@@ -37,25 +38,111 @@ def test_fill_unfill_matrix():
...
+                          Square(Point((i, j)), colors.ASH))
     b.clear_matrix(b.board_tetrominos_matrix)
     for i in range(10):
         for j in range(22):
             assert b.board_tetrominos_matrix[i][j] == 0
 
 
+def test_get_filled_indices_should_be_zero():
+    b = Board(10, 22)
+    assert not b.get_filled_indices()
```
Comment (ctlai95) [inline src/board/board_test.py]: Didn't know about this, this is cool 👍

File: src/board/board.py (hunk 1 above)
Comment (henrytran28) [reply src/board/board.py]: LOL? missing period?

**Comment by henrytran28** [comment]
so many conflicts....

**Comment by henrytran28** [comment]
I couldn't find the bug could you explain how to recreate?

File: src/board/board.py [hunk 6]
```diff
@@ -46,22 +45,73 @@ def render_board(self):
...
+        self.board_tetrominos_squares = board_tetrominos_squares_copy
+
+    def drop_lines(self, indices):
+        """
+        Drops the lines based on the given indices.
+
+        Args:
+            indices (list int): The list of filled indices.
+        """
+        lines_dropped = 0
```
Comment (ctlai95) [inline src/board/board.py]: this var isn't needed either,
can you implement `enumerate` in this for loop instead?
https://stackoverflow.com/questions/522563/accessing-the-index-in-for-loops

File: src/board/board_test.py [hunk 7]
```diff
@@ -65,7 +152,7 @@ def test_hold_current_tetromino():
     b.hold_current_tetromino()
     assert b.held_tetromino.id == last_tetromino_id
     # not dropped yet, so held tetromino should be unchanged
-    b.hold_current_tetromino()
+    b.hold_current_tetromino()
```
Comment (ctlai95) [inline src/board/board_test.py]: remove that whitespace please

File: src/board/board.py [hunk 8]
```diff
@@ -46,22 +45,73 @@ def render_board(self):
...
+        for index in indices:
+            for square in self.board_tetrominos_squares:
+                if square.y > index - lines_dropped:
+                    square.y = square.y - 1
+            lines_dropped += 1
+
+        self.ghost_tetromino = self.get_ghost_tetromino()
```
Comment (ctlai95) [inline src/board/board.py]: Can you put this at the end of movement's hard drop function instead please?

**Comment by henrytran28** [comment]
Yes senpai

**Pull request merged**